quinn-udp/src/unix.rs

```diff
@@ -116,8 +116,9 @@ fn init(io: &std::net::UdpSocket) -> io::Result<()> {
 
     let addr = io.local_addr()?;
 
-    // macos and ios do not support IP_RECVTOS on dual-stack sockets :(
-    if addr.is_ipv4() || ((!cfg!(any(target_os = "macos", target_os = "ios"))) && !only_v6(io)?) {
+    // mac and ios do not support IP_RECVTOS on dual-stack sockets :(
+    // older macos versions also don't have the flag and will error out if we don't ignore it
+    if !only_v6(io)? {
```

Comment:
Probably worth verifying that `only_v6` successfully returns false on a macOS IPv4 socket, rather than failing with a similar weird error.

Reply:
looks like it might be invalid argument on IPv4 socket... so will need to tweak this